### PR TITLE
[12.0][ADD] rjc_sale

### PR DESCRIPTION
**Pull request by dekdeegoodboys**
New Pull request

### CODE REVIEW

File: rjc_sale/models/sale.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.depends('state', 'order_line.invoice_status',
+                 'order_line.invoice_lines')
+    def _get_invoiced(self):
+        super(SaleOrder, self)._get_invoiced()
+        for rec in self:
+            if rec.invoice_status == 'invoiced':
+                picking_status = rec.picking_ids.filtered(lambda l: l.state in ['done',
```
Comment (dekdeegoodboys) [inline rjc_sale/models/sale.py]: Change variable name to **pickings**

Comment (dekdeegoodboys) [inline rjc_sale/models/sale.py]:
```suggestion
                picking_status = rec.picking_ids.filtered(lambda l:
                                                                                l.state in ['done', 'cancel'])
```

File: rjc_sale/models/sale.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+                                                                   'cancel'])
+                if rec.state != 'done' and picking_status:
+                    rec._write({'state': 'done'})
```
Comment (dekdeegoodboys) [inline rjc_sale/models/sale.py]: rec.state = 'done'
or
rec.update({'state': 'done'})

File: rjc_sale/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+    'name': 'RJC Sale',
+    'summary': 'Add condition to hidding lock button and autolock sale order',
+    'version': '12.0.1.0.0',
+    'development_status': 'Beta',
```
Comment (Saran440) [inline rjc_sale/__manifest__.py]:
```suggestion
```

File: rjc_sale/models/sale_order.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    _inherit = 'sale.order'
+
+    @api.depends('state', 'order_line.invoice_status',
+                 'order_line.invoice_lines')
+    def _get_invoiced(self):
+        super(SaleOrder, self)._get_invoiced()
+        for rec in self:
+            if rec.invoice_status == 'invoiced':
+                pickings = rec.picking_ids.filtered(
+                    lambda l: l.state in['done', 'cancel'])
```
Comment (Saran440) [inline rjc_sale/models/sale_order.py]:
```suggestion
                    lambda l: l.state in ['done', 'cancel'])
```

File: rjc_sale/readme/USAGE.rst [hunk 5]
```diff
@@ -0,0 +1,4 @@
+To use this module you need to:
```
Comment (Saran440) [inline rjc_sale/readme/USAGE.rst:1]: Explain how to use this module.

File: rjc_sale/views/sale_order_view.xml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+<odoo>
+    <record id="sale_order_form_inherit_view" model="ir.ui.view">
+        <field name="name">sale.order.form.inherit.view</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//button[@name='action_done' and @states='sale']" position="attributes">
+                <attribute name="states"/>
```
Comment (Saran440) [inline rjc_sale/views/sale_order_view.xml]: `<attribute name="states"/>` what is this code used?

**Pull request merged**